src/utils/helpers/auto-focus/auto-focus.js

```diff
@@ -0,0 +1,24 @@
+import guid from '../guid';
+
+const AutoFocus = {
+  default: (key, autoFocus) => {
+    if (key && autoFocus) {
+      return { key, autoFocus };
+    }
+
+    return {
+      key: guid(),
+      autoFocus: false
+    };
+  },
+
+  getKey: (autoFocus, { ...previous }) => {
```

Comment:
it is in specs for auto-focus.

Reply:
Yes I realise that the second parameter is supplied in these specs:

* https://github.com/blaky/carbon/blob/0cb3002/src/utils/helpers/auto-focus/__spec__.js#L18
* https://github.com/blaky/carbon/blob/0cb3002/src/utils/helpers/auto-focus/__spec__.js#L22

But the specs don't actually affect the runtime behaviour/functionality of Carbon in production.

In all the places where `AutoFocus.getKey()` is called at runtime within the Carbon library, the second parameter is never passed, so it would have no effect on Carbon's runtime behaviour in production:

* https://github.com/blaky/carbon/blob/0cb3002/src/__experimental__/components/checkbox/checkbox.stories.js#L91
* https://github.com/blaky/carbon/blob/0cb3002/src/__experimental__/components/select/select.stories.js#L39
* https://github.com/blaky/carbon/blob/0cb3002/src/__experimental__/components/textarea/textarea.stories.js#L62
* https://github.com/blaky/carbon/blob/0cb3002/src/__experimental__/components/textbox/textbox.stories.js#L150

So is that second parameter implemented for potential future use? (In other words: at some point in the future, maybe a developer might wish to supply that second parameter when writing some new code?)

Reply:
updated.

Reply:
Thanks for adding values for the second parameter :white_check_mark: b19a82d